yktrace/src/hwt/mapper.rs

```diff
@@ -47,25 +52,20 @@ impl HWTMapper {
             //
             // b) `labels` is sorted, so the blocks will be appended to the trace in the
             // correct order.
-            let mut locs = Vec::new();
-            for lbl in &*LABELS {
-                if lbl.off >= start_addr && lbl.off <= end_addr {
-                    // Found matching label.
-                    // Store the virtual address alongside the first basic block, so we can turn
-                    // inlined functions into calls during tracing.
+            let mut locs = LABELS
+                .query(start_addr..(end_addr + 1))
```

Comment:
Why `+ 1`?

Reply:
The library doesn't support inclusive upper-bound ranges for some reason:

```
error[E0308]: mismatched types
  --> yktrace/src/hwt/mapper.rs:56:24
   |
56 |                 .query(start_addr..=end_addr)
   |                        ^^^^^^^^^^^^^^^^^^^^^ expected struct `std::ops::Range`, found struct `RangeInclusive`
   |
   = note: expected struct `std::ops::Range<usize>`
              found struct `RangeInclusive<usize>`

error[E0277]: the trait bound `Element<_, _>: From<(RangeInclusive<usize>, SirLabel)>` is not satisfied
   --> yktrace/src/hwt/mapper.rs:91:5
    |
91  |     IntervalTree::from_iter(vec.into_iter().map(|l| (l.off..=l.off, l)))
    |     ^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(RangeInclusive<usize>, SirLabel)>` is not implemented for `Element<_, _>`
    | 
   ::: /home/vext01/research/yorick/ykrustc/library/core/src/iter/traits/collect.rs:113:34
    |
113 |     fn from_iter<T: IntoIterator<Item = A>>(iter: T) -> Self;
    |                                  -------- required by this bound in `from_iter`
    |
    = help: the following implementations were found:
              <Element<K, V> as From<(std::ops::Range<K>, V)>>
    = note: required because of the requirements on the impl of `Into<Element<_, _>>` for `(RangeInclusive<usize>, SirLabel)`
    = note: required because of the requirements on the impl of `FromIterator<(RangeInclusive<usize>, SirLabel)>` for `IntervalTree<_, _>`
```

Reply:
Ah fair enough.

Reply:
Raised an issue against the library:
https://github.com/main--/rust-intervaltree/issues/8